cluster-install/install.sh

```diff
@@ -2327,55 +2378,58 @@ if components_include_connector; then
 fi
 
 # TRIDENT / ACP yaml
-step_collect_existing_trident_info
-if trident_is_missing; then
-    step_generate_trident_fresh_install_yaml
-elif [ -z "$_EXISTING_TRIDENT_OPERATOR_IMAGE" ]; then
-    logwarn "Upgrading Trident without the Trident Operator is not currently supported, skipping."
-else
-    # Upgrade Trident/Operator?
-    if components_include_trident && ! should_skip_trident_upgrade; then
-        # Trident upgrade (includes operator upgrade if needed)
-        if trident_image_needs_upgraded; then
-            if config_trident_image_is_custom || prompt_user_yes_no "Would you like to upgrade Trident?"; then
-                step_generate_torc_patch "$_EXISTING_TORC_NAME" "$(get_config_trident_image)" "" "" "$(get_config_trident_autosupport_image)"
-                trident_operator_image_needs_upgraded && step_generate_trident_operator_patch
-            else
-                loginfo "Trident will not be upgraded."
-            fi
-        # Trident operator upgrade (standalone)
-        elif trident_operator_image_needs_upgraded; then
-            if config_trident_operator_image_is_custom || prompt_user_yes_no "Would you like to upgrade the Trident Operator?"; then
-                step_generate_trident_operator_patch
-            else
-                loginfo "Trident Operator will not be upgraded."
+step_collect_existing_trident_info && exit_if_problems
+step_existing_trident_flags_compatibility_check && exit_if_problems
```

Comment:
Main changes for this whole Trident-related block is the addition of the "trident flags compatibility" check (i.e. checking if we have a clash between DO_NOT_MODIFY_EXISTING_TRIDENT, COMPONENTS, and an existing Trident install).

The rest is just indentation change from adding the `if trident_will_be_installed_or_modified; then` check on the next line.